src/chainrules.jl

```diff
@@ -193,3 +193,66 @@ ChainRulesCore.@scalar_rule(
 ChainRulesCore.@scalar_rule(expinti(x), exp(x) / x)
 ChainRulesCore.@scalar_rule(sinint(x), sinc(invπ * x))
 ChainRulesCore.@scalar_rule(cosint(x), cos(x) / x)
+
+# non-holomorphic functions
+function ChainRulesCore.frule((_, _, _), ::typeof(besselix), ν::Number, x::Number)
+    Ω = besselix(ν, x)
+    ΔΩ = ChainRulesCore.@not_implemented(BESSEL_ORDER_INFO)
+    return Ω, ΔΩ
+end
+function ChainRulesCore.rrule(::typeof(besselix), ν::Number, x::Number)
+    Ω = besselix(ν, x)
+    project_x = ChainRulesCore.ProjectTo(x)
+    function besselix_pullback(ΔΩ)
+        ν̄ = ChainRulesCore.@not_implemented(BESSEL_ORDER_INFO)
+        a = (besselix(ν - 1, x) + besselix(ν + 1, x)) / 2
+        b = - sign(real(x)) * Ω
+        x̄ = project_x(conj(a) * ΔΩ + real(conj(b) * ΔΩ))
```

Comment:
It's more efficient to compute `-sign(real(x)) * real(conj(Ω) * ΔΩ))` because you're multiplying then 2 reals instead of a real and a complex.

The `@scalar_rule` macro writes rules to use `muladd` here. @oxinabox does that make a difference?

Reply:
I have never measured